src/realm/object-store/sync/sync_session.cpp

```diff
@@ -637,6 +626,23 @@ void SyncSession::handle_error(SyncError error)
                         return; // do not propgate the error to the user at this point
                 }
                 break;
+            case sync::ProtocolErrorInfo::Action::MigrateToFLX:
+                // Should not receive this error if original sync config is FLX
+                REALM_ASSERT(!m_original_sync_config->flx_sync_requested);
+                REALM_ASSERT(error.migration_query_string && !error.migration_query_string->empty());
+                // Original config was PBS, migrating to FLX
+                m_migration_store->migrate_to_flx(*error.migration_query_string);
+                update_sync_config_after_migration();
+                next_state = NextStateAfterError::inactive;
+                break;
+            case sync::ProtocolErrorInfo::Action::RevertToPBS:
+                // Should not receive this error if original sync config is FLX
+                REALM_ASSERT(!m_original_sync_config->flx_sync_requested);
```

Comment:
This is not guaranteed unfortunately. This assertion will fail if the admin reverted the migration after starting to use an FLX config to open their realms.

Reply:
Since we don't have a pbs config anymore, there is no way to rollback unless the server provides the missing data.

Reply:
right, agreed. just wanted to double check if `REALM_ASSERT` is the behavior we want here, since IIUC this will crash the process and tell the user to file a ticket, which would not be useful since this is a known issue. is there a more graceful way to fail here?

also, would `REALM_ASSERT` trigger in a release build?

Reply:
I updated it to use the "switch to pbs" error if this happens and added a test to verify trying to connect as FLX after the server was rolled back.